packages/api/db/migration/20240110204504_create_animal_type_table.js

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_type', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').references('farm_id').inTable('farm').defaultTo(null);
+    table.string('type').defaultTo(null);
+    table.string('type_key').defaultTo(null);
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table.string('created_by_user_id').references('user_id').inTable('users');
+    table.string('updated_by_user_id').references('user_id').inTable('users');
+    table.dateTime('created_at').notNullable();
+    table.dateTime('updated_at').notNullable();
```

Comment:
Should `created_at` and `updated_at` default to now?

Reply:
I'm not sure -- I see we have a mix in our existing migrations between the timestamp above and `now` and no default. Since the model provides timestamps for user inserts, I wonder if the system timestamp is actually the best default? But I don't love that we have no consistency between tables on this.

Reply:
I think system timestamps make sense come to think of it! I'll update my PR to match that

Reply:
Okay I'll do that too 😄